libp2p/NodeTable.cpp

```diff
@@ -74,10 +116,18 @@ list<NodeTable::NodeEntry> NodeTable::state() const
 	return move(ret);
 }
 
-NodeTable::NodeEntry NodeTable::operator[](NodeId _id)
+Node NodeTable::operator[](NodeId _id)
+{
+	Guard l(x_nodes);
+	auto n = m_nodes[_id];
```

Comment:
Would be better to use `find` so that we do not create a new entry at _id if the key is not present.